### PR TITLE
BOLT 1: Reply to unknown channels with an error

### PR DESCRIPTION
**Pull request by pm47**
It seems like a reasonable thing to do but the spec isn't currently clear on that, leading to differences between implementations.

See https://github.com/lightningnetwork/lnd/issues/1882.

### CODE REVIEW

**Comment by pm47** [comment]
> the wording here is super strange

Yeah, I'm afraid I'm not a native speaker :smile:, feel free to propose a better alternative!

> I'd kinda prefer putting a copy of this line in each message in BOLT 2.

I considered that, but we would need to move other parts of BOLT 1 to BOLT 2 for consistency then I think.

**Comment by TheBlueMatt** [comment]
Mostly I just object to the strange wording of "type is between X and Y", which is a weird way to say "all currently-defined messages with types in this range", but that seems like an easy way to end up with self-conflicting specs in the future.

**Comment by btcontract** [comment]
It seems like replying with an `Error` to a channel which does not exists locally kills a use-case where user has the same on-chain wallet (generated from the same seed, hence in many implementations having the same LN node ID) but different channels on many devices. 

Each device will send an `Error` for a channel it does not have (but there is such a channel on another device), as a result all channels for a given node ID will be remotely force-closed if remote peer does automatic closing on remote `Error`.

File: 01-messaging.md [hunk 1]
```diff
@@ -156,6 +156,7 @@ A sending node:
   - when sending `error`:
     - MUST fail the channel referred to by the error message.
   - SHOULD send `error` for protocol violations or internal errors that make channels unusable or that make further communication unusable.
+  - SHOULD send `error` in reply to messages of type `32`-`255` related to unknown channels.
```
Comment (Roasbeef) [inline 01-messaging.md]: SHOULD send _with_ the unknown channel ID

Comment (pm47) [reply 01-messaging.md]: fixed!

**Pull request merged**